Mathlib/LinearAlgebra/FiniteDimensional.lean

```diff
@@ -1381,6 +1393,11 @@ open Module
 
 variable {F E : Type _} [Field F] [Ring E] [Algebra F E]
 
+-- Porting note: these instances can not be found with `etaExperiment`,
+-- so we have to provide a short-circuit instance here.
+instance (S : Subalgebra F E) : AddCommMonoid { x // x ∈ S } := inferInstance
+instance (S : Subalgebra F E) : AddCommGroup { x // x ∈ S } := inferInstance
+
```

Comment:
I think I ran into trouble with these in #3600  too

Reply:
I can remove these if I bump up the heartbeats

Reply:
I removed them; we already need heartbeat hacks in the section, I'd rather avoid a proliferation of instances, as I worry doing that everywhere that will make TC search even worse.

Reply:
I've now added a comment suggesting that we may want to revisit adding these in future.